pandas/tslib.pyx

```diff
@@ -378,11 +378,55 @@ class Timestamp(_Timestamp):
             value = self.tz_localize(None).value
         else:
             value = self.value
+        result = Timestamp(unit*np.round(value/unit),unit='ns')
+        if self.tz is not None:
+            result = result.tz_localize(self.tz)
+        return result
+
+    def floor(self, freq):
+        """
+        return a new Timestamp floored to this resolution
+
+        Parameters
+        ----------
+        freq : a freq string indicating the flooring resolution
+        """
```

Comment:
hmm, not really sure how to do that. you can just repeat the doc-string (but for sure don't repeat code, use a private method for that)